app/src/main/java/com/example/filemanager/ListModel.kt

```diff
@@ -1,7 +1,18 @@
 package com.example.filemanager
 
+import android.os.Environment
 import java.io.File
 
-data class ListModel(
-    val file: File
-)
+data class ListModel(val file: File) {
+    fun isDirectory(): Boolean {
+        return file.isDirectory
+    }
+
+    fun isRootDirectory(): Boolean {
+        return file == Environment.getExternalStorageDirectory()
+    }
+
+    fun parentDirectory(): File? {
+        return file.parentFile
```

Comment:
Wouldn't be better to wrap it in a ListModel so that you wont have to wrap it all the time? The primary model here is the `ListModel`, you should allways operate with it.

Reply:
But it is in ListModel. Could you please explain what do you mean

Reply:
oh, okay, now it's clear, thx